### PR TITLE
2915 rest api documentation

### PR DESCRIPTION
**Pull request by martmull**
- Add a new doc section about Rest API playground (markdown don't render well in dev, don't know why):
image

- Add an input to request user Api Key
image

- Add the Rest API playground
image

### CODE REVIEW

File: packages/twenty-docs/docs/developer/rest_api.mdx [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
 title: Rest API
 sidebar_position: 3
-sidebar_class_name: coming-soon
 sidebar_custom_props:
   icon: TbApi
 ---
 
-Coming soon!
\ No newline at end of file
+To use the Rest API, you will need an API key.
```
Comment (charlesBochet) [inline packages/twenty-docs/docs/developer/rest_api.mdx:8]: REST API

File: packages/twenty-docs/docs/developer/rest_api.mdx [hunk 2]
```diff
@@ -1,9 +1,30 @@
...
+Connect to your Twenty account and follow this doc to get one:
```
Comment (charlesBochet) [inline packages/twenty-docs/docs/developer/rest_api.mdx:9]: follow the following [documentation](link) to generate one

File: packages/twenty-docs/src/pages/graphql.tsx [hunk 3]
```diff
@@ -16,9 +15,7 @@ const GraphiQLComponent = () => {
     window.localStorage.setItem("graphiql:theme", "light");
   }
 
-  const fetcher = createGraphiQLFetcher({
-    url: "https://api.twenty.com/graphql",
-  });
+  const fetcher = createGraphiQLFetcher({url: "http://localhost:3000/graphql"});
```
Comment (charlesBochet) [inline packages/twenty-docs/src/pages/graphql.tsx:18]: oups!

File: packages/twenty-docs/src/pages/rest-api.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
+import Layout from "@theme/Layout";
+import BrowserOnly from "@docusaurus/BrowserOnly";
+import React, { useEffect, useState } from "react";
+import { API } from '@stoplight/elements';
+import '@stoplight/elements/styles.min.css';
+import './graphql.css'
```
Comment (charlesBochet) [inline packages/twenty-docs/src/pages/rest-api.tsx:6]: weird to call it ./graphql.css, should it be ./rest-playground.css?

File: packages/twenty-docs/src/pages/rest-api.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+import { parseJson } from "nx/src/utils/json";
+import { TbLoader2 } from "react-icons/tb";
+
+interface TokenFormPropsI {
```
Comment (charlesBochet) [inline packages/twenty-docs/src/pages/rest-api.tsx:10]: if possible use types and not interfaces per project convention
no need to suffix by I: type TokenFormProps

File: packages/twenty-docs/src/pages/rest-api.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    localStorage.setItem('TryIt_securitySchemeValues', JSON.stringify({bearerAuth: event.target.value}))
+    onSubmit(event.target.value)
+  }
+
+  return !isTokenValid && (
+    <div>
+      <div className='container'>
+      <form className="form">
+        <label>
+          To load all your API endpoints, <a className='link' href='https://app.twenty.com/settings/developers/api-keys'>generate an API key</a> and paste it here:
```
Comment (charlesBochet) [inline packages/twenty-docs/src/pages/rest-api.tsx:28]: To load your REST API schema,

File: packages/twenty-docs/src/pages/rest-api.tsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        </label>
+        <p>
+          <input
+            className={(token && !isLoading) ? "input invalid" : "input"}
+            type='text'
+            disabled={isLoading}
+            placeholder='123'
```
Comment (charlesBochet) [inline packages/twenty-docs/src/pages/rest-api.tsx:35]: let's put a placeholder that looks a bit more like a Bearer (maybe not as long though)

File: packages/twenty-docs/src/pages/rest-api.tsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+const RestApiComponent = () => {
+  const [openApiJson, setOpenApiJson] = useState({})
+  const [isTokenValid, setIsTokenValid] = useState<TokenFormPropsI['isTokenValid']>(undefined)
+  const [isLoading, setIsLoading] = useState(false)
+  const storedToken = parseJson(localStorage.getItem('TryIt_securitySchemeValues'))?.bearerAuth ?? ''
+
+  const validateToken = (openApiJson) => {
+    if(openApiJson.tags) {
```
Comment (charlesBochet) [inline packages/twenty-docs/src/pages/rest-api.tsx:57]: setIsTokenValid(!!openApiJson.tags)

File: packages/twenty-docs/src/pages/rest-api.tsx [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+      setIsTokenValid(true)
+    } else {
+      setIsTokenValid(false)
+    }
+  }
+
+  const getJson = async (token: string ) => {
+    setIsLoading(true)
+    return await fetch(
+      "http://localhost:3000/open-api",
```
Comment (charlesBochet) [inline packages/twenty-docs/src/pages/rest-api.tsx:67]: oups!

File: packages/twenty-docs/src/pages/rest-api.tsx [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+      .then((result)=> {
+        validateToken(result)
+        setIsLoading(false)
+        return result
+      })
+      .catch(() => setIsLoading(false))
+  }
+
+  const submitToken = async (token) => {
+    if(isLoading) return
```
Comment (charlesBochet) [inline packages/twenty-docs/src/pages/rest-api.tsx:80]: if (

File: packages/twenty-docs/src/pages/rest-api.tsx [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+        </p>
+      </form>
+      </div>
+    </div>
+    )
+}
+
+const RestApiComponent = () => {
+  const [openApiJson, setOpenApiJson] = useState({})
+  const [isTokenValid, setIsTokenValid] = useState<TokenFormPropsI['isTokenValid']>(undefined)
```
Comment (charlesBochet) [inline packages/twenty-docs/src/pages/rest-api.tsx:52]: isTokenValid should be a boolean? and initiated with false

**Review by charlesBochet — COMMENTED** [review]
Looks great! We can deploy this to production today! I have left comments

File: packages/twenty-docs/src/pages/graphql.tsx (hunk 3 above)
Comment (martmull) [reply packages/twenty-docs/src/pages/graphql.tsx:18]: well seen

**Review by charlesBochet — APPROVED** [review]
Let's merge it!

**Pull request merged**